.circleci/config.yml

```diff
@@ -61,11 +61,31 @@ commands:
             - node_modules-v3-{{ arch }}-{{ checksum "yarn.lock" }}-{{ checksum "~/.tmp/checksumfiles/patches" }}
       - run:
           name: Extract cached node_modules (if present)
-          command: if [ -e ~/.tmp/node_modules.tgz ]; then tar -xf ~/.tmp/node_modules.tgz; fi
+          command: |
+            if [ -e ~/.tmp/node_modules.tgz ]; then 
+              tar -xf ~/.tmp/node_modules.tgz
+            else
+              # Dependencies have changed, create beacon file used by the install and check licenses steps
+              touch ~/.tmp/yarn_deps_have_changed
```

Comment:
does this get the case where `yarn.lock` was not comitted, but package.json changed?

Reply:
Ah good point! This change will miss the uncommitted `yarn.lock` changes.

In that case, we do need to take the checksum of `package.json` into account for the cache key, so that case is handled too.

Reply:
Issue addressed in https://github.com/celo-org/celo-monorepo/pull/2798/commits/74da4ef52ebc6a3eee1b7b21236c398d7776c9a8

The tradeoff is that the cache will be directly ignored if any of the `package.json` files we have change, even if no dependencies were touched.
But uncommitted changes to `yarn.lock` will be flagged directly without waiting for the next fresh install.

Reply:
nice